staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -119,6 +126,31 @@ func (c *DiscoveryController) sync(version schema.GroupVersion) error {
 			continue
 		}
 		foundVersion = true
+		if c.openAPIAggregationManager != nil {
+			validationSchema, err := getSchemaForVersion(crd, version.Version)
+			if err != nil {
+				return err
+			}
+			// Convert internal CustomResourceValidation to versioned CustomResourceValidation
+			versionedSchema := new(v1beta1.CustomResourceValidation)
+			if validationSchema == nil {
+				versionedSchema = nil
+			} else {
+				if err := v1beta1.Convert_apiextensions_CustomResourceValidation_To_v1beta1_CustomResourceValidation(validationSchema, versionedSchema, nil); err != nil {
+					return err
+				}
+			}
+			crdspec, etag, err := apiextensionsopenapi.CustomResourceDefinitionOpenAPISpec(&crd.Spec, version.Version, versionedSchema)
```

Comment:
Add a comment here that explains we do not to continue even if versionedSchema is nil as it maybe a removal of the schema for this CRD.